### PR TITLE
Added more details in git prompt

### PR DESCRIPTION
**Pull request by edubxb**
- Now, the git repo details are optional through an option: SCM_GIT_SHOW_DETAILS
- The git prompt (only in Powerline and Powerline-plain themes) show the ref type when HEAD not points to a branch:
        tag (tag:v1.0)
        commit (commit:13cb94e)

### CODE REVIEW

**Comment by edubxb** [comment]
Moved functionality from Powerline theme to base theme done. I also updated the README.md.

Check it.

File: themes/powerline-plain/powerline-plain.theme.bash [hunk 1]
```diff
@@ -9,17 +9,7 @@ VIRTUALENV_THEME_PROMPT_COLOR=35
...
-SCM_GIT_UNTRACKED_CHAR="?:"
-SCM_GIT_UNSTAGED_CHAR="U:"
-SCM_GIT_STAGED_CHAR="S:"
-
-if [[ -z "$THEME_SCM_TAG_PREFIX" ]]; then
-    SCM_TAG_PREFIX="tag > "
-else
-    SCM_TAG_PREFIX="$THEME_SCM_TAG_PREFIX"
-fi
+SCM_GIT_SHOW_DETAILS=true
```
Comment (tswicegood) [inline themes/powerline-plain/powerline-plain.theme.bash]: This (and `powerline` below) do not abide by the value set by the user.

File: themes/base.theme.bash [hunk 2]
```diff
@@ -8,9 +8,20 @@ SCM_THEME_PROMPT_DIRTY=' ✗'
 SCM_THEME_PROMPT_CLEAN=' ✓'
 SCM_THEME_PROMPT_PREFIX=' |'
 SCM_THEME_PROMPT_SUFFIX='|'
+SCM_THEME_BRANCH_PREFIX=''
+SCM_THEME_TAG_PREFIX='tag:'
+SCM_THEME_COMMIT_PREFIX='commit:'
+SCM_THEME_REMOTE_PREFIX=''
+
+SCM_GIT_SHOW_DETAILS=${SCM_GIT_SHOW_DETAILS:=false}
```
Comment (tswicegood) [inline themes/base.theme.bash]: I'm inclined to default this to `true` so things don't change for users who are currently expecting the prompt to contain git information.

Comment (edubxb) [inline themes/base.theme.bash]: The only optional information is the working copy status.

Always shown information:

- Branch/Commit/Tag
- Commits ahead
- Commits behind
- Stashed changes

Optional ( visible setting SCM_GIT_SHOW_DETAILS=true):

- Staged files count
- Unstaged files count
- Untracked files count

 anyway, I will change the default value to true.

File: themes/powerline-plain/powerline-plain.theme.bash (hunk 1 above)
Comment (edubxb) [inline themes/powerline-plain/powerline-plain.theme.bash]: The theme follows the user settings as SCM_GIT_SHOW_DETAILS is a global env var, any change done in any global env var after Bash it is loaded is overwritten, so the setting is successfully respected.

**Comment by edubxb** [comment]
I changed the default value of SCM_GIT_SHOW_DETAILS to true, and removed the setting in the powerline themes.

File: themes/powerline-plain/powerline-plain.theme.bash (hunk 1 above)
Comment (tswicegood) [inline themes/powerline-plain/powerline-plain.theme.bash]: Oh!  Gotcha.  Didn't think that one all the way through.

That said, if someone puts the setting before they load bash-it (like many of the settings are) they might get behavior that's not expected.  Probably not a huge deal, but something to keep in mind.

**Comment by tswicegood** [comment]
:+1: Thanks, @edubxb!

Thanks for all the work you've put into Powerline.  It's really amazing!  I've been using it since it was merged in last year and haven't even thought of trying another theme.  It puts such a good face on bash-it, I almost want to make it the default theme. :-)

File: themes/powerline-plain/powerline-plain.theme.bash (hunk 1 above)
Comment (edubxb) [inline themes/powerline-plain/powerline-plain.theme.bash]: You're right, so I removed the setting in Powerline themes ;)

**Comment by edubxb** [comment]
It's a pleasure to contribute to this amazing project! :smile:

**Comment by nwinkler** [comment]
Your contributions are more than welcome - thanks for the great work so far!

**Pull request merged**